hibernate-core/src/main/java/org/hibernate/hql/internal/ast/tree/IdentNode.java

```diff
@@ -197,14 +197,6 @@ private boolean resolveAsAlias() {
 
 		String[] columnExpressions = element.getIdentityColumns();
```

Comment:
So the general idea is to fix this `getIdentityColumns()` to avoid needing the workaround below.

Reply:
Yep. Also, `getIdentityColumns` is only used in `IdentNode#resolveAsAlias`, so changing its behavior should not affect any other node or any other resolution of an `IdentNode`.

Reply:
Can confirm after some scrolling through the code that the replaced logic of `getIdentityColumn` was basically a simplified version of  the now used `toColumns(...)`, with indeed a special handling for subqueries. This change should be fine!

Reply:
After some scrolling through the related code, and having played with `FromElement` recently, I can confirm that the replaced code in `getIdentityColumns()` was indeed a simplified version of what happened in the `toColumns` function now. The latter indeed has a special handling for subqueries. This change looks good to me!